### ISSUE TITLE
How  to  solve "The SonarQube plugin declaring this property should be updated"

### ISSUE THREAD

**Issue by Lenegred** [SonarOpenCommunity/sonar-cxx#1775]
I have met a problem at using ths plugin sonar-c-plugin-1.3.1.1807.jar .

When the jenkins start analyzing prossess,this problem will happen as follow logs.

WARN: Property 'sonar.c.includeDirectories' is not declared as multi-values/property set but was read using 'getStringArray' method. The SonarQube plugin declaring this property should be updated.
WARN: Property 'sonar.c.forceIncludes' is not declared as multi-values/property set but was read using 'getStringArray' method. The SonarQube plugin declaring this property should be updated.
WARN: Property 'sonar.c.cFilesPatterns' is not declared as multi-values/property set but was read using 'getStringArray' method. The SonarQube plugin declaring this property should be updated.

The prossess seized up.

What should I to do ? Logs tell me to update this plugin.But the version 1.3.1 is the lastest.

Thanks.

**Comment by guwirth** [comment]
Hi @Lenegred,

thanks for the feedback. This issue is known and because of SQ API changes. This warnings is to update the plugin code but should have no impact on the functionality. With next plugin release (supports SQ 7.9 LTS only) we have to clean-up/remove all deprecated APIs.

Regards,

**Comment by cheskos** [comment]
I'm still seeing this in Feb 2022 on an Android project with Sonar plugin v3. Is that expected?

**Comment by guwirth** [comment]
@cheskos what do you mean with V3? Which versions of the cxx plugin and SonarQube are you using?

**Comment by atipal** [comment]
I see these warnings too. I have sonarqube plugin version 3.3.

```
Access to the multi-values/property set property 'sonar.java.binaries' should be made using 'getStringArray' method. The SonarQube plugin using this property should be updated. |  
  | Access to the multi-values/property set property 'sonar.java.libraries' should be made using 'getStringArray' method. The SonarQube plugin using this property should be updated. |  
  | Access to the multi-values/property set property 'sonar.java.binaries' should be made using 'getStringArray' method. The SonarQube plugin using this property should be updated. |  
  | Access to the multi-values/property set property 'sonar.java.libraries' should be made using 'getStringArray' method. The SonarQube plugin using this property should be updated.

```

**Comment by guwirth** [comment]
Hi @atipal , is this the right forum. This is the cxx plugin and there is plugin version 3.3?

**Comment by cheskos** [comment]
Hi @guwirth, I think @atipal and me both mean the Android Studio plugin. Apologies if this is the wrong forum for this and would appreciate any pointers.
